internal/services/stock_exchange/algorithms.go

```diff
@@ -0,0 +1,426 @@
+				sellOrderBody.Quantity -= elem.OrderQuantity
+				err := config.DB.Exec("DELETE FROM buy_order_book WHERE id = ?", elem.ID).Error
+				if err != nil {
+					log.Println(err.Error())
+				}
+				err = config.DB.Exec("DELETE FROM sell_order_book WHERE order_id = ?", sellOrderBody.OrderID).Error
+				if err != nil {
+					log.Println(err.Error())
+				}
+			}
+		} else {
+			time.Sleep(30 * time.Second)
+			//Recursive call :
+			SellOrderMatchingAlgo(sellOrderBody, resp)
+		}
+	}
+
+}
+func SellMarketOrder(sellOrderBody OrderRequest, buyBook []model.BuyOrderBook, resp OrderResponse) {
+
+	ltp := 0
+	for _, elem := range buyBook {
+		if elem.OrderQuantity > sellOrderBody.Quantity {
+			ltp = elem.OrderPrice
+			UpdateLTP(ltp, sellOrderBody.StockName)
+			err := config.DB.Exec("UPDATE buy_order_book SET order_quantity = ? , updated_at = ? WHERE id = ? ", elem.OrderQuantity-sellOrderBody.Quantity, time.Now(), elem.ID).Error
+			if err != nil {
+				log.Println(err.Error())
+			}
+			err = config.DB.Exec("DELETE FROM sell_order_book WHERE order_id = ?", resp.OrderID).Error
+			if err != nil {
+				log.Println(err.Error())
+			}
+			SendResponse(resp, "COMPLETED", "Order Executed Partially", resp.OrderID, elem.OrderPrice, sellOrderBody.Quantity)
+			SendResponse(resp, "PARTIAL", "Order Executed Partially", elem.OrderID, elem.OrderPrice, sellOrderBody.Quantity)
+			sellOrderBody.Quantity -= sellOrderBody.Quantity
+			break
+		} else if elem.OrderQuantity < sellOrderBody.Quantity {
+			ltp = elem.OrderPrice
+			UpdateLTP(ltp, sellOrderBody.StockName)
+			SendResponse(resp, "COMPLETED", "Order Executed Partially", elem.OrderID, elem.OrderPrice, elem.OrderQuantity)
+			SendResponse(resp, "PARTIAL", "Order Executed Partially", resp.OrderID, elem.OrderPrice, elem.OrderQuantity)
+			sellOrderBody.Quantity -= elem.OrderQuantity
+			err := config.DB.Exec("DELETE FROM buy_order_book WHERE id = ?", elem.ID).Error
+			if err != nil {
+				log.Println(err.Error())
+			}
+			err = config.DB.Exec("UPDATE sell_order_book SET order_quantity = ? , updated_at = ? WHERE order_id = ?", sellOrderBody.Quantity, time.Now(), sellOrderBody.OrderID).Error
+			if err != nil {
+				log.Println(err.Error())
+			}
+		} else {
+			ltp = elem.OrderPrice
+			UpdateLTP(ltp, sellOrderBody.StockName)
+			SendResponse(resp, "COMPLETED", "Order Executed", resp.OrderID, elem.OrderPrice, sellOrderBody.Quantity)
+			SendResponse(resp, "COMPLETED", "Order Executed", elem.OrderID, elem.OrderPrice, sellOrderBody.Quantity)
+			sellOrderBody.Quantity -= elem.OrderQuantity
+			err := config.DB.Exec("DELETE FROM buy_order_book WHERE id = ?", elem.ID).Error
+			if err != nil {
+				log.Println(err.Error())
+			}
+			err = config.DB.Exec("DELETE FROM sell_order_book WHERE order_id = ?", sellOrderBody.OrderID).Error
+			if err != nil {
+				log.Println(err.Error())
+			}
+		}
+	}
+}
+
+func BuyOrderMatchingAlgo(buyOrderBody OrderRequest, resp OrderResponse) {
```

Comment:
remove resp from function argument here. this can be build while sending response to trading platform.